src/azure-cli/azure/cli/command_modules/feedback/tests/latest/test_feedback.py

```diff
@@ -192,22 +192,18 @@ def _run_cmd(self, command, checks=None, expect_failure=False):
         cli_ctx = get_dummy_cli()
         cli_ctx.logging.command_log_dir = self.temp_command_log_dir
 
-        # hotfix here for untouch feedback's code to avoid introducing possible break change.
-        # unregister the event for auto closing CLI's file logging after execute() is done
-        cli_ctx.unregister_event(EVENT_CLI_POST_EXECUTE, cli_ctx.logging.deinit_cmd_metadata_logging)
-
-        # manually handle error logging as azure.cli.core.util's handle_exception function is mocked out in testsdk / patches
-        # this logged error tests that we can properly parse errors from command log file.
-        with CommandLoggerContext(logger):
-            result = execute(cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
+        # azure.cli.core.util.handle_exception is mocked by azure.cli.testsdk.patches.patch_main_exception_handler
+        # Patch it again so that errors are properly written to command log file.
+        from azure.cli.core.util import handle_exception
+        original_handle_exception = handle_exception
 
-            if result.exit_code != 0:
-                logger.error("There was an error during execution.")
-
-        # close it manually because we unregister the deinit_cmd_metadata_logging
-        # callback from EVENT_CLI_POST_EXECUTE event
-        cli_ctx.logging.end_cmd_metadata_logging(result.exit_code)
+        def _handle_exception_with_log(ex, *args, **kwargs):
+            with CommandLoggerContext(logger):
+                logger.error(ex)
+            original_handle_exception(*args, **kwargs)
 
+        with mock.patch('azure.cli.core.util.handle_exception', _handle_exception_with_log):
+            result = execute(cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
```

Comment:
The old code registers `Logger` as a `Handler`. It works simply because both `Logger` and `Handler` expose `handle` method by coincidence. It is never guaranteed to work as such and may break any time the internal logic of `logging` module changes.